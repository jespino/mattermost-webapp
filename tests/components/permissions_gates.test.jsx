// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
// See License.txt for license information.

import React from 'react';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

import SystemPermissionGate from 'components/permissions_gates/system_permission_gate';
import TeamPermissionGate from 'components/permissions_gates/team_permission_gate';
import ChannelPermissionGate from 'components/permissions_gates/channel_permission_gate';

describe('components/permissions_gates', () => {
    const mockStore = configureStore();
    const state = {
        entities: {
            channels: {
                myMembers: {
                    channel_id: {channel_id: 'channel_id', roles: 'channel_role'}
                }
            },
            teams: {
                myMembers: {
                    team_id: {team_id: 'team_id', roles: 'team_role'}
                }
            },
            users: {
                currentUserId: 'user_id',
                profiles: {
                    user_id: {
                        id: 'user_id',
                        roles: 'system_role'
                    }
                }
            },
            roles: {
                roles: {
                    system_role: {permissions: ['test_system_perm']},
                    team_role: {permissions: ['test_team_perm']},
                    channel_role: {permissions: ['test_channel_perm']}
                }
            }
        }
    };
    const store = mockStore(state);

    describe('SystemPermissionGate', () => {
        test('should match snapshot when user have permission', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <SystemPermissionGate perms={['test_system_perm']}>
                        <p>{'Valid perm (shown)'}</p>
                    </SystemPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
        test('should match snapshot when user have at least on of the permissions', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <SystemPermissionGate perms={['test_system_perm', 'not_existing_perm']}>
                        <p>{'Valid perm (shown)'}</p>
                    </SystemPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
        test('should match snapshot when user have permission and use invert', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <SystemPermissionGate
                        perms={['test_system_perm']}
                        invert={true}
                    >
                        <p>{'Valid perm but inverted (not shown)'}</p>
                    </SystemPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
        test('should match snapshot when user not have permission and use invert', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <SystemPermissionGate
                        perms={['invalid_perm']}
                        invert={true}
                    >
                        <p>{'Invalid perm but inverted (shown)'}</p>
                    </SystemPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
        test('should match snapshot when user haven\'t permission', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <SystemPermissionGate perms={['invalid_perm']}>
                        <p>{'Invalid perm (not shown)'}</p>
                    </SystemPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('TeamPermissionGate', () => {
        test('should match snapshot when user have permission', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <TeamPermissionGate
                        teamId={'team_id'}
                        perms={['test_team_perm']}
                    >
                        <p>{'Valid perm (shown)'}</p>
                    </TeamPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
        test('should match snapshot when user have at least on of the permissions', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <TeamPermissionGate
                        teamId={'team_id'}
                        perms={['test_team_perm', 'not_existing_perm']}
                    >
                        <p>{'Valid perm (shown)'}</p>
                    </TeamPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
        test('should match snapshot when user have permission and use invert', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <TeamPermissionGate
                        teamId={'team_id'}
                        perms={['test_team_perm']}
                        invert={true}
                    >
                        <p>{'Valid perm but inverted (not shown)'}</p>
                    </TeamPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
        test('should match snapshot when user not have permission and use invert', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <TeamPermissionGate
                        teamId={'team_id'}
                        perms={['invalid_perm']}
                        invert={true}
                    >
                        <p>{'Invalid perm but inverted (shown)'}</p>
                    </TeamPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
        test('should match snapshot when user haven\'t permission', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <TeamPermissionGate
                        teamId={'team_id'}
                        perms={['invalid_perm']}
                    >
                        <p>{'Invalid perm (not shown)'}</p>
                    </TeamPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
        test('should match snapshot when the team doesn\'t exists', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <TeamPermissionGate
                        teamId={'invalid_id'}
                        perms={['test_team_perm']}
                    >
                        <p>{'Valid perm invalid team (not shown)'}</p>
                    </TeamPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
        test('should match snapshot when user have permission system wide', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <TeamPermissionGate
                        teamId={'team_id'}
                        perms={['test_system_perm']}
                    >
                        <p>{'Valid perm (shown)'}</p>
                    </TeamPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('ChannelPermissionGate', () => {
        test('should match snapshot when user have permission', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <ChannelPermissionGate
                        channelId={'channel_id'}
                        teamId={'team_id'}
                        perms={['test_channel_perm']}
                    >
                        <p>{'Valid perm (shown)'}</p>
                    </ChannelPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
        test('should match snapshot when user have at least on of the permissions', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <TeamPermissionGate
                        teamId={'team_id'}
                        perms={['test_team_perm', 'not_existing_perm']}
                    >
                        <p>{'Valid perm (shown)'}</p>
                    </TeamPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
        test('should match snapshot when user have permission and use invert', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <ChannelPermissionGate
                        channelId={'channel_id'}
                        teamId={'team_id'}
                        perms={['test_channel_perm']}
                        invert={true}
                    >
                        <p>{'Valid perm but inverted (not shown)'}</p>
                    </ChannelPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
        test('should match snapshot when user not have permission and use invert', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <ChannelPermissionGate
                        channelId={'channel_id'}
                        teamId={'team_id'}
                        perms={['invalid_perm']}
                        invert={true}
                    >
                        <p>{'Invalid perm but inverted (shown)'}</p>
                    </ChannelPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
        test('should match snapshot when user haven\'t permission', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <ChannelPermissionGate
                        channelId={'channel_id'}
                        teamId={'team_id'}
                        perms={['invalid_perm']}
                    >
                        <p>{'Invalid perm (not shown)'}</p>
                    </ChannelPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
        test('should match snapshot when the channel doesn\'t exists', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <ChannelPermissionGate
                        channelId={'invalid_id'}
                        teamId={'team_id'}
                        perms={['test_channel_perm']}
                    >
                        <p>{'Valid perm invalid channel (not shown)'}</p>
                    </ChannelPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
        test('should match snapshot when user have permission team wide', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <ChannelPermissionGate
                        channelId={'channel_id'}
                        teamId={'team_id'}
                        perms={['test_team_perm']}
                    >
                        <p>{'Valid perm (shown)'}</p>
                    </ChannelPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
        test('should match snapshot when user have permission system wide', () => {
            const wrapper = mount(
                <Provider store={store}>
                    <ChannelPermissionGate
                        channelId={'channel_id'}
                        teamId={'team_id'}
                        perms={['test_system_perm']}
                    >
                        <p>{'Valid perm (shown)'}</p>
                    </ChannelPermissionGate>
                </Provider>
            );

            expect(wrapper).toMatchSnapshot();
        });
    });
});
