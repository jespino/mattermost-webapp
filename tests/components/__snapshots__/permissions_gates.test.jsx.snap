// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/permissions_gates ChannelPermissionGate should match snapshot when the channel doesn't exists 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ChannelPermissionGate)
    channelId="invalid_id"
    perms={
      Array [
        "test_channel_perm",
      ]
    }
    teamId="team_id"
  >
    <ChannelPermissionGate
      channelId="invalid_id"
      dispatch={[Function]}
      hasPerm={false}
      invert={false}
      perms={
        Array [
          "test_channel_perm",
        ]
      }
      teamId="team_id"
    />
  </Connect(ChannelPermissionGate)>
</Provider>
`;

exports[`components/permissions_gates ChannelPermissionGate should match snapshot when user have at least on of the permissions 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(TeamPermissionGate)
    perms={
      Array [
        "test_team_perm",
        "not_existing_perm",
      ]
    }
    teamId="team_id"
  >
    <TeamPermissionGate
      dispatch={[Function]}
      hasPerm={true}
      invert={false}
      perms={
        Array [
          "test_team_perm",
          "not_existing_perm",
        ]
      }
      teamId="team_id"
    >
      <p>
        Valid perm (shown)
      </p>
    </TeamPermissionGate>
  </Connect(TeamPermissionGate)>
</Provider>
`;

exports[`components/permissions_gates ChannelPermissionGate should match snapshot when user have permission 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ChannelPermissionGate)
    channelId="channel_id"
    perms={
      Array [
        "test_channel_perm",
      ]
    }
    teamId="team_id"
  >
    <ChannelPermissionGate
      channelId="channel_id"
      dispatch={[Function]}
      hasPerm={true}
      invert={false}
      perms={
        Array [
          "test_channel_perm",
        ]
      }
      teamId="team_id"
    >
      <p>
        Valid perm (shown)
      </p>
    </ChannelPermissionGate>
  </Connect(ChannelPermissionGate)>
</Provider>
`;

exports[`components/permissions_gates ChannelPermissionGate should match snapshot when user have permission and use invert 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ChannelPermissionGate)
    channelId="channel_id"
    invert={true}
    perms={
      Array [
        "test_channel_perm",
      ]
    }
    teamId="team_id"
  >
    <ChannelPermissionGate
      channelId="channel_id"
      dispatch={[Function]}
      hasPerm={true}
      invert={true}
      perms={
        Array [
          "test_channel_perm",
        ]
      }
      teamId="team_id"
    />
  </Connect(ChannelPermissionGate)>
</Provider>
`;

exports[`components/permissions_gates ChannelPermissionGate should match snapshot when user have permission system wide 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ChannelPermissionGate)
    channelId="channel_id"
    perms={
      Array [
        "test_system_perm",
      ]
    }
    teamId="team_id"
  >
    <ChannelPermissionGate
      channelId="channel_id"
      dispatch={[Function]}
      hasPerm={true}
      invert={false}
      perms={
        Array [
          "test_system_perm",
        ]
      }
      teamId="team_id"
    >
      <p>
        Valid perm (shown)
      </p>
    </ChannelPermissionGate>
  </Connect(ChannelPermissionGate)>
</Provider>
`;

exports[`components/permissions_gates ChannelPermissionGate should match snapshot when user have permission team wide 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ChannelPermissionGate)
    channelId="channel_id"
    perms={
      Array [
        "test_team_perm",
      ]
    }
    teamId="team_id"
  >
    <ChannelPermissionGate
      channelId="channel_id"
      dispatch={[Function]}
      hasPerm={true}
      invert={false}
      perms={
        Array [
          "test_team_perm",
        ]
      }
      teamId="team_id"
    >
      <p>
        Valid perm (shown)
      </p>
    </ChannelPermissionGate>
  </Connect(ChannelPermissionGate)>
</Provider>
`;

exports[`components/permissions_gates ChannelPermissionGate should match snapshot when user haven't permission 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ChannelPermissionGate)
    channelId="channel_id"
    perms={
      Array [
        "invalid_perm",
      ]
    }
    teamId="team_id"
  >
    <ChannelPermissionGate
      channelId="channel_id"
      dispatch={[Function]}
      hasPerm={false}
      invert={false}
      perms={
        Array [
          "invalid_perm",
        ]
      }
      teamId="team_id"
    />
  </Connect(ChannelPermissionGate)>
</Provider>
`;

exports[`components/permissions_gates ChannelPermissionGate should match snapshot when user not have permission and use invert 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ChannelPermissionGate)
    channelId="channel_id"
    invert={true}
    perms={
      Array [
        "invalid_perm",
      ]
    }
    teamId="team_id"
  >
    <ChannelPermissionGate
      channelId="channel_id"
      dispatch={[Function]}
      hasPerm={false}
      invert={true}
      perms={
        Array [
          "invalid_perm",
        ]
      }
      teamId="team_id"
    >
      <p>
        Invalid perm but inverted (shown)
      </p>
    </ChannelPermissionGate>
  </Connect(ChannelPermissionGate)>
</Provider>
`;

exports[`components/permissions_gates SystemPermissionGate should match snapshot when user have at least on of the permissions 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(SystemPermissionGate)
    perms={
      Array [
        "test_system_perm",
        "not_existing_perm",
      ]
    }
  >
    <SystemPermissionGate
      dispatch={[Function]}
      hasPerm={true}
      invert={false}
      perms={
        Array [
          "test_system_perm",
          "not_existing_perm",
        ]
      }
    >
      <p>
        Valid perm (shown)
      </p>
    </SystemPermissionGate>
  </Connect(SystemPermissionGate)>
</Provider>
`;

exports[`components/permissions_gates SystemPermissionGate should match snapshot when user have permission 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(SystemPermissionGate)
    perms={
      Array [
        "test_system_perm",
      ]
    }
  >
    <SystemPermissionGate
      dispatch={[Function]}
      hasPerm={true}
      invert={false}
      perms={
        Array [
          "test_system_perm",
        ]
      }
    >
      <p>
        Valid perm (shown)
      </p>
    </SystemPermissionGate>
  </Connect(SystemPermissionGate)>
</Provider>
`;

exports[`components/permissions_gates SystemPermissionGate should match snapshot when user have permission and use invert 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(SystemPermissionGate)
    invert={true}
    perms={
      Array [
        "test_system_perm",
      ]
    }
  >
    <SystemPermissionGate
      dispatch={[Function]}
      hasPerm={true}
      invert={true}
      perms={
        Array [
          "test_system_perm",
        ]
      }
    />
  </Connect(SystemPermissionGate)>
</Provider>
`;

exports[`components/permissions_gates SystemPermissionGate should match snapshot when user haven't permission 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(SystemPermissionGate)
    perms={
      Array [
        "invalid_perm",
      ]
    }
  >
    <SystemPermissionGate
      dispatch={[Function]}
      hasPerm={false}
      invert={false}
      perms={
        Array [
          "invalid_perm",
        ]
      }
    />
  </Connect(SystemPermissionGate)>
</Provider>
`;

exports[`components/permissions_gates SystemPermissionGate should match snapshot when user not have permission and use invert 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(SystemPermissionGate)
    invert={true}
    perms={
      Array [
        "invalid_perm",
      ]
    }
  >
    <SystemPermissionGate
      dispatch={[Function]}
      hasPerm={false}
      invert={true}
      perms={
        Array [
          "invalid_perm",
        ]
      }
    >
      <p>
        Invalid perm but inverted (shown)
      </p>
    </SystemPermissionGate>
  </Connect(SystemPermissionGate)>
</Provider>
`;

exports[`components/permissions_gates TeamPermissionGate should match snapshot when the team doesn't exists 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(TeamPermissionGate)
    perms={
      Array [
        "test_team_perm",
      ]
    }
    teamId="invalid_id"
  >
    <TeamPermissionGate
      dispatch={[Function]}
      hasPerm={false}
      invert={false}
      perms={
        Array [
          "test_team_perm",
        ]
      }
      teamId="invalid_id"
    />
  </Connect(TeamPermissionGate)>
</Provider>
`;

exports[`components/permissions_gates TeamPermissionGate should match snapshot when user have at least on of the permissions 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(TeamPermissionGate)
    perms={
      Array [
        "test_team_perm",
        "not_existing_perm",
      ]
    }
    teamId="team_id"
  >
    <TeamPermissionGate
      dispatch={[Function]}
      hasPerm={true}
      invert={false}
      perms={
        Array [
          "test_team_perm",
          "not_existing_perm",
        ]
      }
      teamId="team_id"
    >
      <p>
        Valid perm (shown)
      </p>
    </TeamPermissionGate>
  </Connect(TeamPermissionGate)>
</Provider>
`;

exports[`components/permissions_gates TeamPermissionGate should match snapshot when user have permission 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(TeamPermissionGate)
    perms={
      Array [
        "test_team_perm",
      ]
    }
    teamId="team_id"
  >
    <TeamPermissionGate
      dispatch={[Function]}
      hasPerm={true}
      invert={false}
      perms={
        Array [
          "test_team_perm",
        ]
      }
      teamId="team_id"
    >
      <p>
        Valid perm (shown)
      </p>
    </TeamPermissionGate>
  </Connect(TeamPermissionGate)>
</Provider>
`;

exports[`components/permissions_gates TeamPermissionGate should match snapshot when user have permission and use invert 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(TeamPermissionGate)
    invert={true}
    perms={
      Array [
        "test_team_perm",
      ]
    }
    teamId="team_id"
  >
    <TeamPermissionGate
      dispatch={[Function]}
      hasPerm={true}
      invert={true}
      perms={
        Array [
          "test_team_perm",
        ]
      }
      teamId="team_id"
    />
  </Connect(TeamPermissionGate)>
</Provider>
`;

exports[`components/permissions_gates TeamPermissionGate should match snapshot when user have permission system wide 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(TeamPermissionGate)
    perms={
      Array [
        "test_system_perm",
      ]
    }
    teamId="team_id"
  >
    <TeamPermissionGate
      dispatch={[Function]}
      hasPerm={true}
      invert={false}
      perms={
        Array [
          "test_system_perm",
        ]
      }
      teamId="team_id"
    >
      <p>
        Valid perm (shown)
      </p>
    </TeamPermissionGate>
  </Connect(TeamPermissionGate)>
</Provider>
`;

exports[`components/permissions_gates TeamPermissionGate should match snapshot when user haven't permission 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(TeamPermissionGate)
    perms={
      Array [
        "invalid_perm",
      ]
    }
    teamId="team_id"
  >
    <TeamPermissionGate
      dispatch={[Function]}
      hasPerm={false}
      invert={false}
      perms={
        Array [
          "invalid_perm",
        ]
      }
      teamId="team_id"
    />
  </Connect(TeamPermissionGate)>
</Provider>
`;

exports[`components/permissions_gates TeamPermissionGate should match snapshot when user not have permission and use invert 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(TeamPermissionGate)
    invert={true}
    perms={
      Array [
        "invalid_perm",
      ]
    }
    teamId="team_id"
  >
    <TeamPermissionGate
      dispatch={[Function]}
      hasPerm={false}
      invert={true}
      perms={
        Array [
          "invalid_perm",
        ]
      }
      teamId="team_id"
    >
      <p>
        Invalid perm but inverted (shown)
      </p>
    </TeamPermissionGate>
  </Connect(TeamPermissionGate)>
</Provider>
`;
